shared/process-tc-artifacts.sh

```diff
@@ -44,6 +44,63 @@ unzip -q $PLATFORM.mozsearch-rust.zip -d objdir-$PLATFORM
 mkdir -p generated-$PLATFORM
 tar -x -z -C generated-$PLATFORM -f $PLATFORM.generated-files.tar.gz
 
+date
+
+# ### Normalize generated source files by rust build-scripts. ###
+#
+# These will look like:
+#   x86_64-unknown-linux-gnu/debug/build/cranelift-codegen-17ba72a9572695af/out/binemit-x86.rs
+# which can be generalized to:
+#   ${RUST_PLATFORM}/${BUILD_TYPE}/build/${CRATE_NAME}-${CRATE_HASH}/out/${FILE_PATH}
+# and which we want to end up with the following path scheme inside this
+# platform specific directory:
+#   __RUST_BUILD_SCRIPT__/${CRATE_NAME}/${FILE_PATH}
+# which will in the end look like (where __linux64__ varies per platform):
+#   __GENERATED__/__linux64__/__RUST_BUILD_SCRIPT__/${CRATE_NAME}/${FILE_PATH}
+#
+# It's also necessary for similar translations to be performed when performing
+# fixups against the save-analysis files.
+
+# Do all of this in the generated-$PLATFORM directory.
+pushd generated-$PLATFORM
+
+declare -A RUST_PLAT_DIRS
+RUST_PLAT_DIRS["linux64"]="x86_64-unknown-linux-gnu"
+RUST_PLAT_DIRS["macosx64"]="x86_64-apple-darwin"
+RUST_PLAT_DIRS["win64"]="x86_64-pc-windows-msvc"
+RUST_PLAT_DIRS["android-armv7"]="thumbv7neon-linux-androideabi"
+RUST_PLATFORM=${RUST_PLAT_DIRS[$PLATFORM]}
+
+function move_file {
+    mkdir -p "$(dirname $2)"
+    mv "$1" "$2"
+}
+
+# Use sed to get a list of all files that match the patterns we specify above
+# and print out ONLY those files, with each resulting line containing the
+# source path followed by a space followed by the target path.  We will then
+# move the files to their target using the `move_file` helper from above which
+# will create the
```

Comment:
... directories as needed? The suspense is killing me!

Reply:
You guessed correctly, although perhaps tautologically, as I went with your suggestion here.  I think that's what I was planning to type before my abrupt context switch.